DataFormats/Portable/test/test_catch2_deepCopyOnHost.cc

```diff
@@ -0,0 +1,126 @@
+#include <Eigen/Core>
+#include <Eigen/Dense>
+
+#include <catch.hpp>
+
+#include "DataFormats/Portable/interface/PortableHostCollection.h"
+#include "DataFormats/SoATemplate/interface/SoACommon.h"
+#include "DataFormats/SoATemplate/interface/SoALayout.h"
+#include "DataFormats/SoATemplate/interface/SoAView.h"
+
+GENERATE_SOA_LAYOUT(SoAPositionTemplate,
+                    SOA_COLUMN(float, x),
+                    SOA_COLUMN(float, y),
+                    SOA_COLUMN(float, z),
+                    SOA_SCALAR(int, detectorType))
+
+using SoAPosition = SoAPositionTemplate<>;
+using SoAPositionView = SoAPosition::View;
+using SoAPositionConstView = SoAPosition::ConstView;
+
+GENERATE_SOA_LAYOUT(SoAPCATemplate,
+                    SOA_COLUMN(float, eigenvalues),
+                    SOA_COLUMN(float, eigenvector_1),
+                    SOA_COLUMN(float, eigenvector_2),
+                    SOA_COLUMN(float, eigenvector_3),
+                    SOA_EIGEN_COLUMN(Eigen::Vector3d, candidateDirection))
+
+using SoAPCA = SoAPCATemplate<>;
+using SoAPCAView = SoAPCA::View;
+using SoAPCAConstView = SoAPCA::ConstView;
+
+GENERATE_SOA_LAYOUT(GenericSoATemplate,
+                    SOA_COLUMN(float, xPos),
+                    SOA_COLUMN(float, yPos),
+                    SOA_COLUMN(float, zPos),
+                    SOA_EIGEN_COLUMN(Eigen::Vector3d, candidateDirection))
+
+using GenericSoA = GenericSoATemplate<>;
+using GenericSoAView = GenericSoA::View;
+using GenericSoAConstView = GenericSoA::ConstView;
+
+TEST_CASE("Deep copy from SoA Generic View") {
+  // common number of elements for the SoAs
+  const std::size_t elems = 10;
+
+  // Portable Collections
+  PortableHostCollection<SoAPosition> positionCollection(elems, cms::alpakatools::host());
+  PortableHostCollection<SoAPCA> pcaCollection(elems, cms::alpakatools::host());
+
+  // Portable Collection Views
+  SoAPositionView& positionCollectionView = positionCollection.view();
+  SoAPCAView& pcaCollectionView = pcaCollection.view();
+  // Portable Collection ConstViews
+  const SoAPositionConstView& positionCollectionConstView = positionCollection.const_view();
+  const SoAPCAConstView& pcaCollectionConstView = pcaCollection.const_view();
+
+  // fill up
+  for (size_t i = 0; i < elems; i++) {
+    positionCollectionView[i] = {i * 1.f, i * 2.0f, i * 3.0f};
```

Comment:
Pedantic: could you use either  `1.0f, 2.0f, ...` or `1.f, 2.f, ...` consistently ?